docs/concepts.md

```diff
@@ -322,6 +322,39 @@ represented by a message containing two Channels handles:
     connection itself; this represents the actual authority of the caller.
   - an empty integrity component.
 
+### HTTP and user labels
+
+Similar to gRPC requests, each incoming HTTP request must have a label
+specifying the confidentiality guarantees that the caller wants to impose on the
+request message. The label is either provided as a
+[JSON formatted](/oak_abi/json/label.json) header named `oak-label` or a
+[protobuf message](/oak_abi/proto/label.proto) in a header named
+`oak-label-bin`.
+
+Internally, each incoming HTTP invocation is represented by a message containing
+two Channel handles:
+
+- a "request receiver" read handle to a Channel whose label is the same as the
+  label provided as a header (i.e., `oak-label` or `oak-label-bin`) in the HTTP
+  request. In a later version this will be updated so that the label on the
+  Channel has:
+
+  - a confidentiality component explicitly specified by the label in the HTTP
+    header; this represents the confidentiality guarantees that the caller wants
+    to impose on the request message.
+  - an integrity component implicitly set by the HTTP server pseudo-Node based
+    on some trusted authentication mechanism that is performed as part of the
+    HTTP connection itself; this represents the actual authority of the caller.
+
+- a "response sender" write handle to a Channel, whose Label is currently
+  `public_untrusted`. However, to avoid any accidental data leaks, the label of
+  the Channel should be set such that it has:
+
+  - a confidentiality component set based on the identity of the user. This is
+    essentially the same as the integrity component described above for the
+    "request receiver".
+  - an empty integrity component.
```

Comment:
How are messages to this channel handled? Does it result in an outgoing HTTP invocation to the client that sent the original invocation?

Reply:
Yes. The messages in the channel are only read by the HTTP server pseudo-node. The messages are in fact HTTP responses that will be sent to the client. But I will add a the confidentiality label to avoid any accidental data leaks. (This is a WIP description of how it should work, and it is not yet implemented, I have created #1428 to track this.)